src/joystick/joystick_manager.cpp

```diff
@@ -0,0 +1,301 @@
+    rtt::Vector2 orientation(orientationY, orientationX);
+
+    // This checks which control mode to use (absolute (FIFA) or relative (Call of Duty))
+    if(!joy.useRelativeControl) {                                               // Checks whether control mode is absolute
+        if (0.9 < orientation.length())                                         // Check whether the orientation stick is pushed far enough
+            joy.orientation = orientation.angle() * 16;                         // Set new orientation
+        command.w = joy.orientation + joy.orientationOffset;                    // Write orientation command and adjust for offset
+    } else {                                                                    // If not, control mode is relative
+        if (0.9 < fabs(orientationX));
+        joy.orientation += orientationX * joy.profile.ROTATION_MULTIPLIER;
+        if (16 * M_PI < joy.orientation)
+            joy.orientation -= 32 * M_PI;       // Bring rotation into range [-16 Pi, 16 Pi]
+        if (joy.orientation < -16 * M_PI)
+            joy.orientation += 32 * M_PI;       // Bring rotation into range [-16 Pi, 16 Pi]
+        command.w = joy.orientation;
+    }
+
+    // Make sure the angle is within range [-16 Pi, 16 Pi]
+    if(16 * M_PI < command.w) command.w -= 32 * M_PI;       // Bring rotation into range [-16 Pi, 16 Pi]
+    if(command.w <-16 * M_PI) command.w += 32 * M_PI;       // Bring rotation into range [-16 Pi, 16 Pi]
+
+    // ==== Set Kicker ====//
+    btn = Xbox360Controller::RightBumper;
+    if(getVal(msg.buttons, xbox360mapping.at(btn)) > 0) {            // If RightBumper is pressed
+        if (!getVal(msg_prev.buttons, xbox360mapping.at(btn)) > 0) { // Check whether it was not already pressed before
+            command.kicker = true;                                  // If not, activate kicker
+            command.kicker_forced = true;                            // Don't wait for ball sensor, kick immediately
+        }
+    }
+
+    /* ==== Set Chipper ====*/
+    double RightTriggerVal = getVal(msg.axes, xbox360mapping.at(Xbox360Controller::RightTrigger));
+    double RightTriggerVal_prev = getVal(msg_prev.axes, xbox360mapping.at(Xbox360Controller::RightTrigger));
+
+    // The RightTriggerVal is a range from -1 to 1, where -1 is fully pressed and 1 is fully unpressed
+    if(RightTriggerVal <= -0.9) {
+        if(RightTriggerVal_prev > -0.9) {
+            std::cout<<"CHIP!"<<std::endl;
+            command.chipper = true;
+            command.chipper_forced = true;
+        }
+    }
+    /* ==================== */
+
+    // ==== Set kicker velocity
+    if(command.kicker || command.chipper) {
+        command.kicker_vel = 5.0;
+        command.chipper_vel = 5.0;
+    }
+
+    // ==== Set dribbler ====
+    btn = Xbox360Controller::LeftBumper;
+    if(getVal(msg.buttons, xbox360mapping.at(btn))){            // Check whether the Left Bumper is pressed
```

Comment:
No comments on the right side of statements such that the line becomes so long. think about small screens :)